fvm/evm/types/events.go

```diff
@@ -123,7 +123,7 @@ func (p *BlockExecutedEventPayload) CadenceEvent() (cadence.Event, error) {
 
 	fields := []cadence.Value{
 		cadence.NewUInt64(p.Block.Height),
-		cadence.NewUInt64(p.Block.TotalSupply),
+		cadence.String(p.Block.TotalSupply.String()),
```

Comment:
Maybe just use `Int` here?

```suggestion
		cadence.NewIntFromBig(p.Block.TotalSupply),
```

In any case, the type definition for the event, `blockExecutedEventCadenceType`, needs to be adjusted accordingly

Reply:
Nice, didn't know Cadence supports big.Int under the hood for ints larger than the uint64.

Reply:
updated to big.Int, also fixed the order